[1.6.2] Standalone server crash on BiomeEventHandler, possible forge bug?

As far as I can tell there's nothing wrong with the BiomeEventHandler class. GetWaterColor is a client only event, but TFC isn't trying to register it. Seems like a forge bug, I tried forge 827 and it still fails. Maybe someone else can figure out how to fix it.

```
java.lang.NoClassDefFoundError: net/minecraftforge/event/terraingen/BiomeEvent$GetWaterColor
    at java.lang.Class.getDeclaringClass(Native Method)
    at java.lang.Class.getEnclosingClass(Unknown Source)
    at java.lang.Class.getSimpleBinaryName(Unknown Source)
    at java.lang.Class.getSimpleName(Unknown Source)
    at net.minecraftforge.event.ASMEventHandler.getUniqueName(ASMEventHandler.java:111)
    at net.minecraftforge.event.ASMEventHandler.createWrapper(ASMEventHandler.java:59)
    at net.minecraftforge.event.ASMEventHandler.<init>(ASMEventHandler.java:28)
    at net.minecraftforge.event.EventBus.register(EventBus.java:77)
    at net.minecraftforge.event.EventBus.register(EventBus.java:58)
    at TFC.TerraFirmaCraft.initialize(TerraFirmaCraft.java:177)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:540)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at com.google.common.eventbus.EventHandler.handleEvent(EventHandler.java:74)
    at com.google.common.eventbus.SynchronizedEventHandler.handleEvent(SynchronizedEventHandler.java:45)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:313)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:296)
    at com.google.common.eventbus.EventBus.post(EventBus.java:267)
    at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:194)
    at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:174)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at com.google.common.eventbus.EventHandler.handleEvent(EventHandler.java:74)
    at com.google.common.eventbus.SynchronizedEventHandler.handleEvent(SynchronizedEventHandler.java:45)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:313)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:296)
    at com.google.common.eventbus.EventBus.post(EventBus.java:267)
    at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:105)
    at cpw.mods.fml.common.Loader.initializeMods(Loader.java:697)
    at cpw.mods.fml.server.FMLServerHandler.finishServerLoading(FMLServerHandler.java:94)
    at cpw.mods.fml.common.FMLCommonHandler.onServerStarted(FMLCommonHandler.java:355)
    at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:141)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:443)
    at net.minecraft.server.ThreadMinecraftServer.run(SourceFile:582)
Caused by: java.lang.ClassNotFoundException: net.minecraftforge.event.terraingen.BiomeEvent$GetWaterColor
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:179)
    at java.lang.ClassLoader.loadClass(Unknown Source)
    at java.lang.ClassLoader.loadClass(Unknown Source)
    ... 42 more
Caused by: java.lang.RuntimeException: Attempted to load class net/minecraftforge/event/terraingen/BiomeEvent$GetWaterColor for invalid side SERVER
    at cpw.mods.fml.common.asm.transformers.SideTransformer.transform(SideTransformer.java:50)
    at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:267)
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:165)
    ... 44 more
```

--- a/TFC_Shared/src/TFC/ClientProxy.java
+++ b/TFC_Shared/src/TFC/ClientProxy.java
@@ -24,6 +24,7 @@ import net.minecraft.tileentity.TileEntity;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.BiomeGenBase;
+import net.minecraftforge.client.GuiIngameForge;
 import net.minecraftforge.common.MinecraftForge;
 import TFC.API.Enums.EnumTree;
 import TFC.Core.ColorizerFoliageTFC;
@@ -75,6 +76,7 @@ import TFC.GUI.GuiSluice;
 import TFC.GUI.GuiVessel;
 import TFC.GUI.GuiVesselLiquid;
 import TFC.GUI.GuiWorkbench;
+import TFC.Handlers.BiomeEventHandler;
 import TFC.Handlers.Client.BlockRenderHandler;
 import TFC.Handlers.Client.ChiselHighlightHandler;
 import TFC.Handlers.Client.FarmlandHighlightHandler;
@@ -214,6 +216,23 @@ public class ClientProxy extends CommonProxy
 
 		//Register our overlay changes
 		MinecraftForge.EVENT_BUS.register(new RenderOverlayHandler());
+	}
+	
+	@Override
+	@SideOnly(Side.CLIENT)
+	public void registerBiomeEventHandler()
+	{
+		// Register the Biome Event Handler
+		MinecraftForge.EVENT_BUS.register(new BiomeEventHandler());
+	}
+	
+	@Override
+	@SideOnly(Side.CLIENT)
+	public void setupGuiIngameForge()
+	{
+		GuiIngameForge.renderHealth = false;
+		GuiIngameForge.renderArmor = false;
+		GuiIngameForge.renderFood = false;
 	}
 
 	@Override

--- a/TFC_Shared/src/TFC/CommonProxy.java
+++ b/TFC_Shared/src/TFC/CommonProxy.java
@@ -57,6 +57,7 @@ import TFC.Entities.Mobs.EntitySpiderTFC;
 import TFC.Entities.Mobs.EntitySquidTFC;
 import TFC.Entities.Mobs.EntityWolfTFC;
 import TFC.Entities.Mobs.EntityZombieTFC;
+import TFC.Handlers.BiomeEventHandler;
 import TFC.TileEntities.TEBlastFurnace;
 import TFC.TileEntities.TECrucible;
 import TFC.TileEntities.TileEntityAnvil;
@@ -94,6 +95,16 @@ public class CommonProxy implements IGuiHandler
 {
 
 	public void registerRenderInformation() {
+		// NOOP on server
+	}
+	
+	public void registerBiomeEventHandler()
+	{
+		// NOOP on server
+	}
+	
+	public void setupGuiIngameForge()
+	{
 		// NOOP on server
 	}
 

--- a/TFC_Shared/src/TFC/TerraFirmaCraft.java
+++ b/TFC_Shared/src/TFC/TerraFirmaCraft.java
@@ -121,10 +121,6 @@ public class TerraFirmaCraft
 	@EventHandler
 	public void initialize(FMLInitializationEvent evt)
 	{
-		GuiIngameForge.renderHealth = false;
-		GuiIngameForge.renderArmor = false;
-		GuiIngameForge.renderFood = false;
-
 		//Add Item Name Localizations
 		Localization.addLocalization("/assets/terrafirmacraft/lang/", "en_US");
 		//LanguageRegistry.instance().loadLocalization("assets/terrafirmacraft/lang/", "en_US", false);
@@ -174,9 +170,10 @@ public class TerraFirmaCraft
 
 		//Register our player tracker
 		GameRegistry.registerPlayerTracker(new PlayerTracker());
-
-		// Register the Biome Event Handler
-		MinecraftForge.EVENT_BUS.register(new BiomeEventHandler());
+		
+		proxy.registerBiomeEventHandler();
+		
+		proxy.setupGuiIngameForge();
 
 		//Setup custom potion effects
 		TFCPotion.Setup();

